server/controllers/user.js

```diff
@@ -21,3 +21,89 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
 
   res.status(200).json({ users: users });
 });
+
+// @route PUT /users/:username/follow
+// @desc Follow a user
+// @access Private
+exports.followUser = asyncHandler(async (req, res, next) => {
+  const current_username = req.body.username;
+  const other_username = req.params.username;
+
+  if (other_username !== current_username) {
```

Comment:
Rather than an if else, you could use a single if statement with an early return statement to avoid the nesting structure. Something like `return res.status(403)`